0000-catalog.md

```diff
@@ -0,0 +1,781 @@
+- If a catalog_product with catalog_product_id does not exist the transaction
+  is invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+  The owner in the product must match the organization that the agent belongs to,
+  otherwise the transaction is invalid.
+- The agent must have the permission can_update_product for the organization,
+  otherwise the transaction is invalid.
+- The properties must be valid for the catalog_product schema; its properties
+  must only contain properties that are included in the catalog_product Schema.
+
+The inputs for CatalogProductUpdateAction must include:
+
+- Address of the Agent submitting the transaction
+- Address of the Organization the Product is being updated for
+- Address of the Product to be updated
+
+The outputs for CatalogProductUpdateAction must include:
+
+- Address of the updated catalog_product
+
+### CatalogProductDeleteAction
+
+CatalogProductDeleteAction removes an existing catalog_product from state. The
+transaction should be submitted by an agent, identified by its signing key, 
+acting on behalf of the organization that corresponds to the org_id in the 
+product being deleted. (Organizations and agents are defined by the Pike smart 
+contract.)
+
+```
+message CatalogProductDeleteAction {
+    // catalog_id and product_id are used in deriving the state address
+    string catalog_id = 1;
+    string product_id = 2;
+}
+```
+
+If the Grid setting grid.product.allow_delete is set to false, this transaction
+is invalid (sys admin setting). The default value for grid.product.allow_delete 
+is true. This setting is stored using the Sawtooth Settings smart contract, more 
+information can be found [here](https://sawtooth.hyperledger.org/docs/core/releases/1.0/transaction_family_specifications/settings_transaction_family.html).
+
+Validation requirements:
+
+- If a catalog_product with catalog_product_id does not exist the transaction is
+  invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The owner in the product must match the organization that the agent belongs
+  to, otherwise the transaction is invalid.
+- The agent must have the permission “can_delete_product” for the organization
+  otherwise the transaction is invalid.
+
+The inputs for CatalogProductDeleteAction must include:
+
+- Address of the Agent submitting the transaction
+- Address of the Organization the Product is being deleted for
+- Address of the catalog_product to be deleted
+
+The outputs for CatalogProductDeleteAction must include:
+
+- Address of the catalog_product to be deleted
+
+**_NOTE: Deleting a catalog_product is potentially dangerous operation that
+could leave dangling references and should be done with care._**
+
+## Catalog_Product Operations
+
+### CatalogProductSetStatus
```

Comment:
This section should be updated to explain that it can update the the status of multiple Catalog Products. For example it should explain that every product must be able to be updated or the transaction is invalid,  and change inputs and outputs to be for the multiple products and so on.

Reply:
✅

Reply:
✅

Reply:
I might be missing the update for the above comment but I don't see the RFC contents reflect the request. I have a follow on to the item @agunde406 brings up and that is:

```
if an attempt to update multiple Catalog Products in 
one transaction is requested, should the RFC reflect
the expectation to apply the outcome at the whole 
transaction level or it is acceptable to handle each Product 
individually?
```